src/Uno.UI/UI/Xaml/Documents/Hyperlink.cs

```diff
@@ -202,18 +230,28 @@ internal void OnClick()
 			}
 #endif
 		}
+		#endregion
 
-		private Brush GetPressedForeground()
+		private void SetCurrentForeground()
 		{
-#if XAMARIN
-			var normalColor = Brush.GetColorWithOpacity(Foreground, Colors.Transparent).Value;
-			var pressedColor = Color.FromArgb((byte)(normalColor.A / 2), normalColor.R, normalColor.G, normalColor.B);
-			return new SolidColorBrush(pressedColor);
-#else
-			return null;
-#endif
+			if (_pressedState.pointer is { }
+				&& _pressedState.themes.UnoGetValueOrDefault(HyperlinkForegroundPressedKey,null) is Brush brush)
+			{
+				this.SetValue(ForegroundProperty, brush, DependencyPropertyValuePrecedences.Animations);
+			}
+			else
+			{
+				if (_hoveredState.pointer is { }
```

Comment:
This should probably be outside the `else`. That is, check for **both** `_pressedState` and `_hoveredState`. I *think* it might be possible to have both of them set (e.g, multi-pointers)?

I'm not very sure though, but it's safest to check for both anyways.

Reply:
Well, only one value at Precedence=Animations can be set at any time, so even if we have 2 pointers -one pressed and one hovering- we still have to pick 1. Pressed is stronger, so it takes priority.